src/ApplicationInsights.Kubernetes/TelemetryInitializers/SimplePerformanceCounterTelemetryInitializer.cs

```diff
@@ -0,0 +1,77 @@
+#if NETSTANDARD2_0
+
+using System;
+using System.Diagnostics;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using static Microsoft.ApplicationInsights.Kubernetes.TelemetryProperty;
+
+namespace Microsoft.ApplicationInsights.Kubernetes
+{
+    internal sealed class SimplePerformanceCounterTelemetryInitializer : ITelemetryInitializer
+    {
+        private TimeSpan _lastCPUSample;
+        private Stopwatch _cpuWatch = new Stopwatch();
+
+        // Cache current process for performance trade off. 
+        // Refer to https://stackoverflow.com/questions/28023490/should-one-call-dispose-for-process-getcurrentprocess
```

Comment:
I think it is OK to not bother with disposing the Process object because `SimplePerformanceCounterTelemetryInitializer` is used as a **singleton** (one per Application Insights telemetry processor chain). That is a better justification IMO than relying on some Process implementation details.

Reply:
Thanks for bring this up. I am sorry but I only partially agree :-).

I agree rely on the callee's implementation details is not ideal.

However, IMHO, it is as not ideal as rely on caller's implementation - there is NO constraint saying that `SimplePerformanceCounterTelemetryInitializer` or any `ITelemetryInitializer` have to be used as a singleton either.

That said, this crossed my mind, I just think it too heavy to implement for now. Post it here totally for discussion:
Basically, abstract out the current process part.
To create a singleton `CurrentProcessorInfoProvider`, that provides a method to get info from the current process. Implementation wise, current process object will be lazy load and cached, and it could implement IDisposable.

Let me know what do you think. :-D

Reply:
I like your proposal a lot, with just one caveat: as far as I can tell, the property getters of the `Process` class are not thread-safe, so the provider code will probably have to ensure that only one thread is accessing the `Process` instance at any given time

Reply:
Task for another day. TBH, at this moment, I am more willing to rule out the SimplePerformanceCounterTelemetryInitializer. Here are my reasons:
1. **It is NOT k8s specific, and doesn't have to be in this repository;** (shall I say it shouldn't be here?)
1. .NET Core 3.x provides performance counters, I think it will be way more reliable to use that as the implementation to get metrics like CPU/Memory usage.

Reply:
All that said, thank you so much for the review!